src/arweave/composite-client.ts

```diff
@@ -635,29 +635,51 @@ export class ArweaveCompositeClient
   async getData({ id }: { id: string }): Promise<ContiguousData> {
     this.failureSimulator.maybeFail();
 
-    const [dataResponse, dataSizeResponse] = await Promise.all([
-      this.trustedNodeRequestQueue.push({
-        method: 'GET',
-        url: `/tx/${id}/data`,
-      }),
-      this.trustedNodeRequestQueue.push({
-        method: 'GET',
-        url: `/tx/${id}/data_size`,
-      }),
-    ]);
+    try {
+      const [dataResponse, dataSizeResponse] = await Promise.all([
+        this.trustedNodeRequestQueue.push({
+          method: 'GET',
+          url: `/tx/${id}/data`,
+        }),
+        this.trustedNodeRequestQueue.push({
+          method: 'GET',
+          url: `/tx/${id}/data_size`,
+        }),
+      ]);
 
-    if (!dataResponse.data) {
-      throw Error('No transaction data');
-    }
+      if (!dataResponse.data) {
+        throw Error('No transaction data');
+      }
 
-    const size = +dataSizeResponse.data;
-    const txData = fromB64Url(dataResponse.data);
-    return {
-      stream: Readable.from(txData),
-      size,
-      verified: false,
-      cached: false,
-    };
+      const size = +dataSizeResponse.data;
+      const txData = fromB64Url(dataResponse.data);
+
+      const stream = Readable.from(txData);
+
+      stream.on('error', () => {
+        metrics.getDataStreamErrorsTotal.inc({
+          class: 'ArweaveCompositeClient',
```

Comment:
I *think* we can use `this.constructor.name` to get the class name (might have to be outside the callback).

Reply:
Great suggestion!